### PR TITLE
fix(x11/python-torch): include numpy files when building (fixes #21744)

### LINKED ISSUE #21744: [Bug]: python-torch wasnt compiled with numpy support
### Problem description

for a few days ive been getting an error on code which used to run fine before

```
/data/data/com.termux/files/usr/lib/python3.12/site-packages/transformers/image_processing_utils.py:41: UserWarning: The following named arguments are not valid for `ViTImageProcessor.preprocess` and were ignored: 'padding'
  return self.preprocess(images, **kwargs)
Loading from kha-white/manga-ocr-base... Traceback (most recent call last):
  File "/data/data/com.termux/files/usr/lib/python3.12/site-packages/transformers/feature_extraction_utils.py", line 193, in convert_to_tensors
    tensor = as_tensor(value)
             ^^^^^^^^^^^^^^^^
  File "/data/data/com.termux/files/usr/lib/python3.12/site-packages/transformers/feature_extraction_utils.py", line 150, in as_tensor
    return torch.from_numpy(value)
           ^^^^^^^^^^^^^^^^^^^^^^^
RuntimeError: PyTorch was compiled without NumPy support
```

i suspect [this commit](https://github.com/termux/termux-packages/commit/01c8e30d532f42108846272d4e24d607d0264af8) is causing the error in which [this line](https://github.com/termux/termux-packages/blob/6d7a9aa9fd39d670f1045322c97d4831108c9447/x11-packages/python-torch/build.sh#L23) was changed

### What steps will reproduce the bug?

1. install python-torch and python-numpy
```sh
pkg i python-pytorch python-numpy
```
2. try using torch.from_numpy function in your code
```sh
python3 -c 'import torch; torch.from_numpy([])'
```

### What is the expected behavior?

it doesnt give any error

### System information

```shell
Termux Variables:
TERMUX_APP_PACKAGE_MANAGER=pacman
TERMUX_APP__APK_FILE=/data/app/~~uxuy_8yusUgdkjYnksT-vw==/com.termux-yNiatvZRy63EY-qnF9dfjw==/base.apk
TERMUX_APP__APK_RELEASE=GITHUB
TERMUX_APP__APP_VERSION_CODE=1020
TERMUX_APP__APP_VERSION_NAME=0.119.0-beta.1
TERMUX_APP__DATA_DIR=/data/user/0/com.termux
TERMUX_APP__IS_DEBUGGABLE_BUILD=true
TERMUX_APP__IS_INSTALLED_ON_EXTERNAL_STORAGE=false
TERMUX_APP__PACKAGE_NAME=com.termux
TERMUX_APP__PID=26128
TERMUX_APP__TARGET_SDK=28
TERMUX_VERSION=0.119.0-beta.1
TERMUX__SE_FILE_CONTEXT=u:object_r:app_data_file:s0:c70,c257,c512,c768
TERMUX__SE_INFO=default:targetSdkVersion=28:complete
TERMUX__SE_PROCESS_CONTEXT=u:r:untrusted_app_27:s0:c70,c257,c512,c768
TERMUX__UID=10326
TERMUX__USER_ID=0
Packages CPU architecture:
aarch64
Subscribed repositories:
# /data/data/com.termux/files/usr/etc/pacman.conf
[main]
Usage = All
Server = https://service.termux-pacman.dev/main/aarch64
Server = https://s3.amazonaws.com/termux-pacman.us/main/aarch64
[x11]
Usage = All
Server = https://service.termux-pacman.dev/x11/aarch64
Server = https://s3.amazonaws.com/termux-pacman.us/x11/aarch64
[root]
Usage = All
Server = https://service.termux-pacman.dev/root/aarch64
Server = https://s3.amazonaws.com/termux-pacman.us/root/aarch64
[tur]
Usage = All
Server = https://service.termux-pacman.dev/tur/aarch64
Server = https://s3.amazonaws.com/termux-pacman.us/tur/aarch64
[tur-continuous]
Usage = All
Server = https://service.termux-pacman.dev/tur-continuous/aarch64
Server = https://s3.amazonaws.com/termux-pacman.us/tur-continuous/aarch64
[tur-multilib]
Usage = All
Server = https://service.termux-pacman.dev/tur-multilib/aarch64
Server = https://s3.amazonaws.com/termux-pacman.us/tur-multilib/aarch64
[gpkg]
Usage = All
Server = https://service.termux-pacman.dev/gpkg/aarch64
Server = https://s3.amazonaws.com/termux-pacman.us/gpkg/aarch64
Updatable packages:
cmake 3.30.4-0 -> 3.30.5-0
exiftool 12.97-0 -> 12.98-0
eza 0.20.1-0 -> 0.20.2-0
git 2.46.2-0 -> 2.47.0-0
lua-language-server 3.11.0-0 -> 3.11.1-0
opencv 4.10.0-2 -> 4.10.0-3
python-torchvision 0.19.1-0 -> 0.19.1-1
texlab 5.19.0-0 -> 5.20.0-0
uv 0.4.18-0 -> 0.4.20-0
termux-tools version:
1.43.6
Android version:
14
Kernel build information:
Linux localhost 4.19.191-28577532-abA346EXXU8CXG8 #1 SMP PREEMPT Wed Jul 31 10:33:19 KST 2024 aarch64 Android
Device manufacturer:
samsung
Device model:
SM-A346E
LD Variables:
LD_LIBRARY_PATH=
LD_PRELOAD=/data/data/com.termux/files/usr/lib/libtermux-exec.so
Installed termux plugins:
com.termux.widget versionCode:13
com.termux.x11 versionCode:15
com.termux.styling versionCode:1000
com.termux.api versionCode:51
```

### PR DESCRIPTION
**Pull request by AminurAlam**
see https://github.com/termux/termux-packages/issues/21744

### CODE REVIEW

**Review by fornwall — CHANGES_REQUESTED** [review]
This doesn't work - NumPy support is still not built, check the build log:

> 2024-10-10T14:13:30.6174431Z   NumPy could not be found.  Not building with NumPy.  Suppress this warning
> 2024-10-10T14:13:30.6175343Z   with -DUSE_NUMPY=OFF

We should probably patch away this fallback, to instead fail the build if not building the desired numpy support - https://github.com/pytorch/pytorch/blob/main/cmake/Dependencies.cmake#L886-L892:

```cmake
    if(USE_NUMPY)
      if(NOT Python_NumPy_FOUND)
        message(WARNING "NumPy could not be found. Not building with NumPy. Suppress this warning with -DUSE_NUMPY=OFF")
        caffe2_update_option(USE_NUMPY OFF)
      else()
        caffe2_update_option(USE_NUMPY ON)
      endif()
```

**Comment by fornwall** [comment]
Relevant change in pytorch regarding the problem: https://github.com/pytorch/pytorch/commit/8ea1dc874880de9bbcf32699c4098a3c5e3f3b20, `Use Python::NumPy target`.

File: x11-packages/python-torch/build.sh [hunk 1]
```diff
@@ -20,7 +20,7 @@ TERMUX_PKG_EXTRA_CONFIGURE_ARGS="
 -DCMAKE_BUILD_TYPE=Release
 -DCMAKE_INSTALL_PREFIX=${TERMUX_PKG_SRCDIR}/torch
 -DCMAKE_PREFIX_PATH=${TERMUX_PYTHON_HOME}/site-packages
--DNumPy_INCLUDE_DIR=${TERMUX_PYTHON_HOME}/site-packages/numpy/_core/include
+-DNUMPY_INCLUDE_DIR=${TERMUX_PYTHON_HOME}/site-packages/numpy/_core/include
```
Comment (licy183) [inline x11-packages/python-torch/build.sh:23]:
```suggestion
-DPython_NumPy_INCLUDE_DIR=${TERMUX_PYTHON_HOME}/site-packages/numpy/_core/include
```

**Comment by AminurAlam** [comment]
i have added a patch for fail and renamed the flag
```diff
-        message(WARNING "NumPy could not be found. Not building with NumPy. Suppress this warning with -DUSE_NUMPY=OFF")
+        message(FATAL_ERROR "NumPy could not be found. Not building with NumPy. Suppress this warning with -DUSE_NUMPY=OFF")
```

**Review by fornwall — APPROVED** [review]

**Comment by fornwall** [comment]
Nice work, seems to work beautifully now!

**Pull request merged**